doto-backend/src/models/Task.js

```diff
@@ -2,15 +2,15 @@ const mongoose = require('mongoose');
 const Schema = mongoose.Schema;
 
 const taskSchema = mongoose.Schema({
+    id: {
+        type: Schema.Types.ObjectId,
+        // required: true
+    },
     user: {
         type: Schema.Types.ObjectId,
         ref: 'user',
         required: true
     },
-    taskId: {
-        type: String,
-        required: true
-    },
     title: {
         type: String
     },
```

Comment:
Wouldn't it make more sense for the title of a task to be required?

Reply:
heya, you don't need to include the id in the schema, it is assumed already :)

Reply:
wdym it's assumed? Like it's auto generated?

Reply:
yeah, it doesn't need to be specified, it's auto-generated and can be accessed through "_id"

Reply:
@lucycy98 by assumed do you mean that in the front end, the entry to the title is a mandatory entry box where if it's blank, then the user can't submit the task?

Reply:
> yeah, it doesn't need to be specified, it's auto-generated and can be accessed through "_id"

Oh right this makes sense. Never mind about my above comment.

Reply:
_id is a MongoDB specific ID to represent an object. I think each object should still have it's own unique ID separate from the one in MongoDB. Usually when APIs return unique data, the data includes a unique ID.

Reply:
I'm not sure whether it's 'bad practice' to use the MongoDB id as the id of the object, I'm just assuming based on what I've seen in the past. I don't think it's a big deal, up to you :P

Reply:
Hmm so how is the ID meant to be generated? Just not too sure why we would need two IDs haha

Reply:
You would return the non Mongo ID in the API response, so the user of the API can uniquely identify documents.

https://stackoverflow.com/questions/105034/create-guid-uuid-in-javascript

Reply:
From what I've read online, the Mongo IDs aren't generated sequentially and are effectively meant to be like a guid. So I think it'll be fine to use them instead of having to generate our own unique ids.

Reply:
Aight, then I think we can merge this change?

Reply:
Oh, one more thing, sorry for being so pedantic, but wouldn't the time zone (at the end of the timedate) be important?

Reply:
> Oh, one more thing, sorry for being so pedantic, but wouldn't the time zone (at the end of the timedate) be important?

ISO-8601 time format uses an offset at the end for time zone. I think we can omit this for the time being and add it as a "later feature" if we're thinking of using different time zones as a feature. Otherwise, we'd just assume its local to NZ standard time.

Reply:
> > Oh, one more thing, sorry for being so pedantic, but wouldn't the time zone (at the end of the timedate) be important?
> 
> ISO-8601 time format uses an offset at the end for time zone. I think we can omit this for the time being and add it as a "later feature" if we're thinking of using different time zones as a feature. Otherwise, we'd just assume its local to NZ standard time.

When we generate dates, e.g. using toISOString(), it adds the timezone on the end automatically, unless you mean something else?

Reply:
> > > Oh, one more thing, sorry for being so pedantic, but wouldn't the time zone (at the end of the timedate) be important?
> > 
> > 
> > ISO-8601 time format uses an offset at the end for time zone. I think we can omit this for the time being and add it as a "later feature" if we're thinking of using different time zones as a feature. Otherwise, we'd just assume its local to NZ standard time.
> 
> When we generate dates, e.g. using toISOString(), it adds the timezone on the end automatically, unless you mean something else?

Isn't our time in the format "2020-08-14T08:50:00" atm? It doesn't seem to have the time zone at the end, or am I mistaken. It seems like that's what we're using for the tests.

Reply:
Yea, what I meant was that it **should** have the timezone at the end, even in the tests

Reply:
> Yea, what I meant was that it **should** have the timezone at the end, even in the tests

I'll change the tests to include timezone, thanks for pointing it out 😄